src/v2/components/AttachmentPicker/AttachmentPicker.tsx

```diff
@@ -0,0 +1,308 @@
+  attachmentPickerBottomSheetHandleHeight?: number;
+  attachmentPickerBottomSheetHeight?: number;
+  attachmentPickerErrorButtonText?: string;
+  attachmentPickerErrorText?: string;
+  numberOfAttachmentImagesToLoadPerCall?: number;
+  numberOfAttachmentPickerImageColumns?: number;
+};
+
+export const AttachmentPicker = React.forwardRef(
+  (props: AttachmentPickerProps, ref: React.ForwardedRef<BottomSheet>) => {
+    const {
+      AttachmentPickerBottomSheetHandle,
+      attachmentPickerBottomSheetHandleHeight,
+      attachmentPickerBottomSheetHeight,
+      AttachmentPickerError,
+      attachmentPickerErrorButtonText,
+      AttachmentPickerErrorImage,
+      attachmentPickerErrorText,
+      ImageOverlaySelectedComponent,
+      numberOfAttachmentImagesToLoadPerCall,
+      numberOfAttachmentPickerImageColumns,
+    } = props;
+
+    const {
+      theme: {
+        attachmentPicker: { bottomSheetContentContainer },
+      },
+    } = useTheme();
+    const {
+      maxNumberOfFiles,
+      selectedImages,
+      selectedPicker,
+      setSelectedImages,
+      setSelectedPicker,
+      topInset,
+    } = useAttachmentPickerContext();
+
+    const [appState, setAppState] = useState<AppStateStatus>();
+    const [currentIndex, setCurrentIndex] = useState(-1);
+    const [endCursor, setEndCursor] = useState<string>();
+    const [photoError, setPhotoError] = useState(false);
+    const [hasNextPage, setHasNextPage] = useState(true);
+    const [loadingPhotos, setLoadingPhotos] = useState(false);
+    const [photos, setPhotos] = useState<string[]>([]);
+
+    const hideAttachmentPicker = () => {
+      setSelectedPicker(undefined);
+      if ((ref as React.MutableRefObject<BottomSheet>)?.current) {
+        (ref as React.MutableRefObject<BottomSheet>).current.close();
+      }
+    };
+
+    const getMorePhotos = async () => {
+      if (hasNextPage && !loadingPhotos) {
+        setLoadingPhotos(true);
+        try {
+          const results = await getPhotos({
+            after: endCursor,
+            first: numberOfAttachmentImagesToLoadPerCall ?? 60,
+          });
+          setEndCursor(results.endCursor);
+          setHasNextPage(results.hasNextPage || false);
+          setPhotos([...photos, ...results.assets]);
+        } catch (error) {
+          console.log(error);
+          setPhotoError(true);
+        }
+        setLoadingPhotos(false);
+      }
+    };
+
+    useEffect(() => {
+      const handleAppStateChange = (nextAppState: AppStateStatus) => {
+        setAppState(nextAppState);
+      };
+      AppState.addEventListener('change', handleAppStateChange);
+
+      return () => {
+        AppState.removeEventListener('change', handleAppStateChange);
+      };
+    }, []);
+
+    useEffect(() => {
+      if (Platform.OS === 'ios') {
+        Keyboard.addListener('keyboardWillShow', hideAttachmentPicker);
+      } else {
+        Keyboard.addListener('keyboardDidShow', hideAttachmentPicker);
```

Comment:
do we need to make sure clients are using a particular `android:windowSoftInputMode` in their Android manifest? if so, we should make sure we document that